pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfigUtils.java

```diff
@@ -373,6 +373,14 @@ public static FunctionConfig convertFromDetails(FunctionDetails functionDetails)
 
         functionConfig.setCleanupSubscription(functionDetails.getSource().getCleanupSubscription());
         functionConfig.setAutoAck(functionDetails.getAutoAck());
+
+        // Set subscription position
+        if (Function.SubscriptionPosition.EARLIEST.equals(functionDetails.getSource().getSubscriptionPosition())) {
+            functionConfig.setSubscriptionPosition(SubscriptionInitialPosition.Earliest);
+        } else {
+            functionConfig.setSubscriptionPosition(SubscriptionInitialPosition.Latest);
+        }
```

Comment:
What is the reason for this conditional? Is it because `getSubscriptionPosition` can return `null`? If we ensure that the `getSubscriptionPosition` defaults to `Latest`, we could skip this conditional and just have:

```java
functionConfig.setSubscriptionPosition(functionDetails.getSource().getSubscriptionPosition());
```

Reply:
We cannot directly set FunctionConfig's `SubscriptionPosition` to proto's `SubscriptionPosition`, needs to have a type conversion. So we can have a conversion util function and reuse the code, PTAL at the latest commit, thanks.

Reply:
Thank you for the context. I hadn't noticed that nuance.